EIPS/eip-7530.md

```diff
@@ -0,0 +1,139 @@
+
+### Hardfork Reference Point
+
+The baseline for the opcodes and precompiles defined in this specification is the Shanghai Hark Fork. All opcodes and precompiles not specified in the sections below MUST be fully supported.
+
+### EVM Profile A
+
+This profile describes a system that has minimal changes compared to Ethereum.
+
+#### Opcodes with Changed Behaviour
+
+##### Opcode 0x44: DIFFICULTY / PREVRANDAO
+
+The DIFFICULTY opcode MUST return 0x00 instead of the previous RANDAO value.
+
+#### Precompiles with Changed Behaviour
+
+None.
+
+
+### EVM Profile B
+
+This profile describes a system that inherits the changes of EVM Profile A, plus additionally does not support some precompiles.
+
+#### Opcodes with Changed Behaviour
+
+##### Opcode 0x44: DIFFICULTY / PREVRANDAO
+
+The DIFFICULTY opcode MUST return 0x00 instead of the previous RANDAO value.
+
+##### Precompiles with Changed Behaviour
+
+##### Address 0x02: SHA256
+
+The SHA256 precompile MUST be treated as a revert, returning all gas to the previous context and setting the success flag to 0x00.
+
+##### Address 0x03: RIPEMD160
+
+the RIPEMD160 precompile must be treated as a revert, returning all gas to the previous context and setting the success flag to 0x00.
+
+##### Address 0x05: MODEXP
+
+The MODEXP must be treated as a revert, returning all gas to the previous context and setting the success flag to 0x00.
+
+##### Address 0x06: ALT BN128 ADD
+
+The ALT BN128 ADD precompile must be treated as a revert, returning all gas to the previous context and setting the success flag to 0x00.
+
+##### Address 0x07: ALT BN128 MUL
+
+The ALT BN128 MUL precompile must be treated as a revert, returning all gas to the previous context and setting the success flag to 0x00.
+
+##### Address 0x08: ALT BN128 PAIRING CHECK
+
+The ALT BN128 PAIRING CHECK precompile must be treated as a revert, returning all gas to the previous context and setting the success flag to 0x00.
+
+##### Address 0x09: BLAKE2F
+
+The BLAKE2F precompile must be treated as a revert, returning all gas to the previous context and setting the success flag to 0x00.
+
+### Summary of Profiles
+
+The tables below show a summary of opcodes and precompiles with changed behaviour.
+
+| Opcode                         | Profile A  | Profile B |
+|--------------------------------|------------|-----------|
+| 0x44: DIFFICULTY / PREVRANDAO  | Returns 0  | Returns 0 |
+
+| Precompiles                    | Profile A  | Profile B |
+|--------------------------------|------------|-----------|
+| 0x02: SHA256                   | No change  | Reverts   |
+| 0x03: RIPEMD160                | No change  | Reverts   |
+| 0x05: MODEXP                   | No change  | Reverts   |
+| 0x06: ALT BN128 ADD            | No change  | Reverts   |
+| 0x07: ALT BN128 MUL            | No change  | Reverts   |
+| 0x08: ALT BN128 PAIRING CHECK  | No change  | Reverts   |
+| 0x09: BLAKE2F                  | No change  | Reverts   |
+
+
+## Rationale
+
+TBD
```

Comment:
i think the mentioned profiles need explanation/reasoning to be having those opcodes with changed behavior